src/xunit.netcore.extensions/Discoverers/ConditionalFactDiscoverer.cs

```diff
@@ -23,45 +24,71 @@ public ConditionalFactDiscoverer(IMessageSink diagnosticMessageSink) : base(diag
         public override IEnumerable<IXunitTestCase> Discover(
             ITestFrameworkDiscoveryOptions discoveryOptions, ITestMethod testMethod, IAttributeInfo factAttribute)
         {
-            MethodInfo testMethodInfo = testMethod.Method.ToRuntimeMethod();
+            string[] conditionMemberNames = factAttribute.GetConstructorArguments().FirstOrDefault() as string[];
+            IEnumerable<IXunitTestCase> testCases = base.Discover(discoveryOptions, testMethod, factAttribute);
+            return DiscoverConditionalTestCases(discoveryOptions, _diagnosticMessageSink, testMethod, testCases, conditionMemberNames);
+        }
 
-            string conditionMemberName = factAttribute.GetConstructorArguments().FirstOrDefault() as string;
-            Type declaringType = testMethodInfo.DeclaringType;
-            string[] symbols = conditionMemberName.Split('.');
+        // This helper method evaluates the given condition member names for a given set of test cases.
+        // If any condition member evaluates to 'false', the test cases are marked to be skipped.
+        // The skip reason is the collection of all the condition members that evalated to 'false'.
+        internal static IEnumerable<IXunitTestCase> DiscoverConditionalTestCases(
+                                                        ITestFrameworkDiscoveryOptions discoveryOptions,
+                                                        IMessageSink diagnosticMessageSink,
+                                                        ITestMethod testMethod,
+                                                        IEnumerable<IXunitTestCase> testCases,
+                                                        IEnumerable<string> conditionMemberNames)
+        {
+            MethodInfo testMethodInfo = testMethod.Method.ToRuntimeMethod();
+            Type testMethodDeclaringType = testMethodInfo.DeclaringType;
+            List<string> falseConditions = new List<string>();
 
-            if (symbols.Length == 2)
+            foreach (string entry in conditionMemberNames)
```

Comment:
Good point -- I'll add a check.  In fact, null strings in the list could also be a problem and worth checking.